plugins/woocommerce/client/blocks/assets/js/blocks/product-filters/inner-blocks/taxonomy-filter/edit.tsx

```diff
@@ -22,37 +26,130 @@ import type { FilterOptionItem } from '../../types';
 import { InitialDisabled } from '../../components/initial-disabled';
 import { Notice } from '../../components/notice';
 import { getTaxonomyLabel } from './utils';
+import { sortFilterOptions } from '../../utils/sort-filter-options';
 
 const Edit = ( props: EditProps ) => {
 	const { attributes: blockAttributes } = props;
 
-	const { taxonomy, isPreview, displayStyle, showCounts, sortOrder } =
-		blockAttributes;
+	const {
+		taxonomy,
+		isPreview,
+		displayStyle,
+		showCounts,
+		sortOrder,
+		hideEmpty,
+	} = blockAttributes;
 
-	// For now, use fake data - will be replaced with real taxonomy data later
 	const [ termOptions, setTermOptions ] = useState< FilterOptionItem[] >(
 		[]
 	);
 	const [ isOptionsLoading, setIsOptionsLoading ] =
 		useState< boolean >( true );
 
+	// Fetch taxonomy terms using WordPress core data
+	const { taxonomyTerms, isTermsLoading } = useSelect(
+		( select ) => {
+			if ( ! taxonomy || isPreview ) {
+				return { taxonomyTerms: [], isTermsLoading: false };
+			}
+
+			const { getEntityRecords, hasFinishedResolution } =
+				select( coreStore );
+			const selectorArgs = [
+				'taxonomy',
+				taxonomy,
+				{
+					per_page: -1,
```

Comment:
I suggest limiting it to 15 (the amount that is shown by default. `-1` is fine if we expose the data to user to choose from (e.g. in inspector controls). There's no point to fetch all the categories if we only display 15 in the preview.

```suggestion
					per_page: 15,
```

Reply:
Addressed in [b2ec68d](https://github.com/woocommerce/woocommerce/pull/59609/commits/b2ec68df7ec290f6388d1e69bf5449b50417afbf)